pkg/server/server.go

```diff
@@ -60,7 +68,23 @@ func (s *Server) signHandler(w http.ResponseWriter, r *http.Request) {
 
 	signature, err := s.Signer.Sign(r.Context(), keyHash, data)
 	if err != nil {
-		s.logger().Errorf("Error signing request: %v", err)
+		p, _ := s.Signer.GetPublicKey(r.Context(), keyHash)
```

Comment:
This PR includes changes from your other PR. It might be better to make your new branch of your modified branch.

Reply:
Agreed. Next time I will create a branch in case of making new changes.